File: higherlower/app/src/main/java/com/example/higherlower/MainActivity.java
package com.example.higherlower;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    int n;
    public void clickme(View view)
    {

        String message;
        EditText editText=(EditText)findViewById(R.id.editText);
        int guessno=Integer.parseInt(editText.getText().toString());
        if (guessno>n)
        { message= "go lower";}
        else if (guessno<n)
        { message="go higher";}
        else
            message="yayy, you got it!";
        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
        Log.i("info","pressed");

    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Random rand=new Random();
        n=rand.nextInt(50)+1;
    }
}

File: listViews/app/src/main/java/com/example/listviews/MainActivity.kt
package com.example.listviews

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.list_item_fruit.view.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        lvfruit.adapter = ArrayAdapter(this,R.layout.list_item_fruit, R.id.fruitname,arrayOf("apple","mango","guava","straw","BERRY"))
        lvfruit.setOnItemClickListener { parent, view, position, id ->
            Toast.makeText(this, "Johnny ate $position ${view.fruitname.text}", Toast.LENGTH_LONG).show()
        }
        button.setOnClickListener {
            val i=Intent(this,MainActivity2::class.java)
            startActivity(i)
        }
    }
}